File: frontend/src/api/devtools.ts
import { apiClient } from './client'

export interface TradingStateResetPayload {
  confirmation: string
  initial_cash?: number | null
}

export interface TradingStateResetResult {
  account_id: number
  tenant_id: string
  account_name: string
  initial_cash: string
  available_cash: string
  total_equity: string
  deleted_counts: Record<string, number>
}

export async function resetTradingState(payload: TradingStateResetPayload): Promise<TradingStateResetResult> {
  const { data } = await apiClient.post('/devtools/reset-trading-state', payload)
  return data
}

File: backend/app/schemas/devtools.py
from decimal import Decimal

from pydantic import BaseModel, Field


class TradingStateResetRequest(BaseModel):
    confirmation: str = Field(description="Must be RESET")
    initial_cash: Decimal | None = Field(default=None, ge=0)


class TradingStateResetRead(BaseModel):
    account_id: int
    tenant_id: str
    account_name: str
    initial_cash: Decimal
    available_cash: Decimal
    total_equity: Decimal
    deleted_counts: dict[str, int]


class SystemBenchmarkRead(BaseModel):
    status: str
    generated_at: str
    samples: dict[str, dict[str, float | int | str]]
    summary: dict[str, float | int | str]
